### PR TITLE
Make yaml features optional

### PR DESCRIPTION
**Pull request by lafrech**
Addresses part of #246.

TODO:

- [x] Update docs
- [x] Update changelog

Points to check:

setup.py: I added `'yaml'` in `extra_requires`

In `APISpec` class, yaml stuff is imported when calling `to_yaml` and `ImportError` is not caught:

```python
    def to_yaml(self):
        from .yaml_utils import dict_to_yaml
        return dict_to_yaml(self.to_dict())
```

### CODE REVIEW

**Comment by lafrech** [comment]
I forgot to mention that this PR is based on https://github.com/marshmallow-code/apispec/pull/249.

Also, regarding `APISpec.to_yaml`, there are alternatives to importing `yaml_utils` in the method:

- Remove the method and let the user do that manually (import `dict_to_yaml` and call `to_dict`).
- Propose an `APISpecYAML` mixin in `yaml_utils`.

But adding a warning in `APISpec.to_yaml` docstring should be enough.

**Review by sloria — APPROVED** [review]
LGTM. @lafrech Can you update this with `dev`? Once that's done, go ahead an merge this.

**Comment by lafrech** [comment]
I think it is already up-to-date with `dev`. It includes the commits in #249.

The docs might need an update, though. Can't promise when I'll do it (summer holidays).

**Comment by sloria** [comment]
No prob; I've gone ahead and updated the docs and changelog.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,10 @@ Features:
 - [apispec.core]: *Backwards-incompatible*: Remove `Path` class.
   Plugins' `path_helper` methods should now return a path as a string
   and optionally mutate the `operations` dictionary (:pr:`238`).
+- [apispec.core]: *Backwards-incompatible*: YAML support is optional. To
+  install with YAML support, use ``pip install 'apispec[yaml]'``. You
+  will need to do this if you use the ``apispec.ext.flask`` or
+  ``apispec.ext.tornado`` plugins.
```
Comment (lafrech) [inline CHANGELOG.rst]: `FlaskPlugin`, `TornadoPlugin`, or `BottlePlugin`

Comment (sloria) [reply CHANGELOG.rst]: Good catch; fixed.

**Pull request merged**